inductiva/fluids/scenarios/coastal_area/bathymetry.py

```diff
@@ -180,14 +180,40 @@ def from_random_depths(
 
         return cls(depths.flatten(), x.flatten(), y.flatten())
 
-    def to_text_file(self, text_file_path: str):
-        """Writes the bathymetry to a text file.
+    def to_bot_file(
+        self,
+        bot_file_path: str,
+    ):
+        """Writes the bathymetry to a bot file.
+
+        The depth values are interpolated to a regular grid and written to a bot
+        file. The grid size is determined automatically from the range and the
+        number of unique x and y values.
 
         Args:
             text_file_path: Path to the text file.
         """
 
-        np.savetxt(text_file_path, self.depths)
+        x_grid, y_grid = inductiva.utils.grids.get_meshgrid(
+            x_range=self.x_range,
+            y_range=self.y_range,
+            x_num=len(self.x_uniques()),
+            y_num=len(self.y_uniques()),
+        )
+
+        depths_grid = inductiva.utils.interpolation.interpolate_to_uniform_grid(
+            x=(self.x, self.y),
+            values=self.depths,
+            x_grid=(x_grid, y_grid),
+        )
+
+        if np.sum(np.isnan(depths_grid)) > 0:
```

Comment:
To clarify, users will still need to be aware of this?

The idea is that they clip before where they want or to just interpolate those values as you explained before. Is this the workflow you are thinking?

Reply:
Yep. If they attempt to use an invalid bathymetry, they are informed. They should crop the bathymetry (with a method that I plan to add later today) until it is valid for simulations.